spectrum-network/src/peer/peer_store.rs

```diff
@@ -0,0 +1,57 @@
+use libp2p::PeerId;
+use std::collections::HashMap;
+use crate::peer::data::{PeerInfo};
+
+pub enum PeerStoreRejection {
+    StoreExhausted,
+    AlreadyExists,
+}
+
+pub trait PeerStore {
+    fn add(&mut self, peer_id: &PeerId, peer: PeerInfo) -> Result<&mut PeerInfo, PeerStoreRejection>;
+    fn drop(&mut self, peer_id: &PeerId) -> bool;
+    fn get(&self, peer_id: &PeerId) -> Option<&PeerInfo>;
+    fn get_mut(&mut self, peer_id: &PeerId) -> Option<&mut PeerInfo>;
+}
+
+pub struct PeerStoreConfig {
+    capacity: usize,
+}
+
+pub struct InMemoryPeerStore {
+    peers: HashMap<PeerId, PeerInfo>,
+    conf: PeerStoreConfig,
+}
+
+impl PeerStore for InMemoryPeerStore {
+    fn add(&mut self, peer_id: &PeerId, peer: PeerInfo) -> Result<&mut PeerInfo, PeerStoreRejection> {
```

Comment:
`PeerId` is a `Copy` type. I would  rather remove the reference, or use `*peer_id` when needed